agent/consul/rpc_client.go

```diff
@@ -84,23 +90,38 @@ func (c *RPCClient) grpcConn(server *metadata.Server) (*grpc.ClientConn, error)
 	host, _, _ := net.SplitHostPort(server.Addr.String())
 	addr := fmt.Sprintf("%s:%d", host, server.Port)
 
-	c.grpcConnsLock.RLock()
-	conn, ok := c.grpcConns[addr]
-	c.grpcConnsLock.RUnlock()
-	if ok {
-		return conn, nil
-	}
+	var isLead bool
```

Comment:
can you change lead to isPendingConnection to something similar (lead is misleading in Consul context)